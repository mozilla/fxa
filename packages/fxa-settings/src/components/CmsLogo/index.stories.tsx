/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { Meta } from '@storybook/react';
import CmsLogo from '.';
import { withLocalization } from 'fxa-react/lib/storybooks';

export default {
  title: 'Components/CmsLogo',
  component: CmsLogo,
  decorators: [
    withLocalization,
    (Story) => (
      <div className="flex flex-col gap-4 p-10 max-w-lg">
        <Story />
      </div>
    ),
  ],
} as Meta;

// Sample logo data for stories
const mockMozillaLogo = {
  logoUrl: 'https://gist.githubusercontent.com/dschom/857ebb2abd5f75937f211f1fd6bbf9a8/raw/33037c8905757a594e07eb29818d8519447fdec0/nightly-logo.svg',
  logoAltText: 'Mozilla',
};

const mockFirefoxLogo = {
  logoUrl: 'https://gist.githubusercontent.com/dschom/c754b76333cda59f50845fe9b0ff6d52/raw/157b0af53142b24e0ddae936c711c010428e7bba/foxy-logo.svg',
  logoAltText: 'Firefox',
};

export const DesktopLeftAligned = () => (
  <CmsLogo
    isMobile={false}
    logoPosition="left"
    logos={[mockMozillaLogo]}
  />
);

export const DesktopCenterAligned = () => (
  <CmsLogo
    isMobile={false}
    logoPosition="center"
    logos={[mockMozillaLogo]}
  />
);

export const MobileHidden = () => (
  <CmsLogo
    isMobile={true}
    logoPosition="left"
    logos={[mockMozillaLogo]}
  />
);

export const NoLogo = () => (
  <CmsLogo
    isMobile={false}
    logoPosition="left"
    logos={[]}
  />
);

export const InvalidLogo = () => (
  <CmsLogo
    isMobile={false}
    logoPosition="left"
    logos={[{ logoUrl: undefined, logoAltText: undefined }]}
  />
);

export const MultipleLogos = () => (
  <CmsLogo
    isMobile={false}
    logoPosition="center"
    logos={[
      { logoUrl: undefined, logoAltText: undefined }, // Invalid logo (will be skipped)
      mockFirefoxLogo, // This should be used
      mockMozillaLogo, // This will be ignored (first valid logo is used)
    ]}
  />
);

export const FirefoxLogo = () => (
  <CmsLogo
    isMobile={false}
    logoPosition="center"
    logos={[mockFirefoxLogo]}
  />
);

export const Comparison = () => (
  <div className="space-y-8">
    <div>
      <h3 className="text-lg font-semibold mb-4">Left Aligned (Desktop)</h3>
      <CmsLogo
        isMobile={false}
        logoPosition="left"
        logos={[mockMozillaLogo]}
      />
    </div>

    <div>
      <h3 className="text-lg font-semibold mb-4">Center Aligned (Desktop)</h3>
      <CmsLogo
        isMobile={false}
        logoPosition="center"
        logos={[mockMozillaLogo]}
      />
    </div>

    <div>
      <h3 className="text-lg font-semibold mb-4">Mobile (Hidden)</h3>
      <p className="text-sm text-gray-600 mb-2">Logo is hidden on mobile devices</p>
      <CmsLogo
        isMobile={true}
        logoPosition="left"
        logos={[mockMozillaLogo]}
      />
    </div>

    <div>
      <h3 className="text-lg font-semibold mb-4">No Logo</h3>
      <p className="text-sm text-gray-600 mb-2">When no valid logo is provided</p>
      <CmsLogo
        isMobile={false}
        logoPosition="center"
        logos={[]}
      />
    </div>
  </div>
);
