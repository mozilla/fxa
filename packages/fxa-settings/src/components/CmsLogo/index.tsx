/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';

const CmsLogo = (opts: {
  isMobile: boolean;
  logoPosition: 'left' | 'center';
  logos: Array<
    | {
        logoUrl?: string;
        logoAltText?: string;
      }
    | undefined
  >;
}) => {
  const logo = opts.logos.find(
    (x) => x?.logoAltText != null && x?.logoUrl != null
  );
  return (
    <>
      {!opts.isMobile && logo && (
        <img
          src={logo.logoUrl}
          alt={logo.logoAltText}
          className={`${
            opts.logoPosition === 'center' ? 'justify-center mb-4 max-h-[160px]' : 'justify-left mb-4 max-h-[40px]'
          }`}
        />
      )}
    </>
  );
};

export default CmsLogo;
