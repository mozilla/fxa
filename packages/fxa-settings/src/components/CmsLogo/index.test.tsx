/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { render, screen } from '@testing-library/react';
import CmsLogo from '.';

describe('CmsLogo', () => {
  it('renders correctly with default left positioning', () => {
    const { container } = render(
      <CmsLogo
        {...{
          isMobile: false,
          logoPosition: 'left',
          logos: [
            {
              logoAltText: 'foo',
              logoUrl: '/foo.svg',
            },
          ],
        }}
      />
    );

    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
    expect(img.getAttribute('alt')).toEqual('foo');
    expect(img.getAttribute('src')).toEqual('/foo.svg');
    expect(img).toHaveClass('justify-left');
    expect(img).toHaveClass('max-h-[40px]');
    expect(img).not.toHaveClass('justify-center');
    expect(img).not.toHaveClass('max-h-[160px]');
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with center positioning', () => {
    const { container } = render(
      <CmsLogo
        {...{
          isMobile: false,
          logoPosition: 'center',
          logos: [
            {
              logoAltText: 'foo',
              logoUrl: '/foo.svg',
            },
          ],
        }}
      />
    );

    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
    expect(img.getAttribute('alt')).toEqual('foo');
    expect(img.getAttribute('src')).toEqual('/foo.svg');
    expect(img).toHaveClass('justify-center');
    expect(img).toHaveClass('max-h-[160px]');
    expect(img).not.toHaveClass('justify-left');
    expect(img).not.toHaveClass('max-h-[40px]');
    expect(container).toMatchSnapshot();
  });

  it('falls back when alt text is missing', () => {
    render(
      <CmsLogo
        {...{
          isMobile: false,
          logoPosition: 'left',
          logos: [
            {
              logoAltText: undefined,
              logoUrl: '/foo.svg',
            },
            {
              logoAltText: 'bar',
              logoUrl: '/bar.svg',
            },
          ],
        }}
      />
    );

    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
    expect(img.getAttribute('alt')).toEqual('bar');
    expect(img.getAttribute('src')).toEqual('/bar.svg');
  });

  it('falls back when url is missing', () => {
    render(
      <CmsLogo
        {...{
          isMobile: false,
          logoPosition: 'left',
          logos: [
            {
              logoAltText: 'foo',
              logoUrl: undefined,
            },
            {
              logoAltText: 'bar',
              logoUrl: '/bar.svg',
            },
          ],
        }}
      />
    );

    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
    expect(img.getAttribute('alt')).toEqual('bar');
    expect(img.getAttribute('src')).toEqual('/bar.svg');
  });

  it('falls back when logo state is missing', () => {
    render(
      <CmsLogo
        {...{
          isMobile: false,
          logoPosition: 'left',
          logos: [
            undefined,
            {
              logoAltText: 'bar',
              logoUrl: '/bar.svg',
            },
          ],
        }}
      />
    );

    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
    expect(img.getAttribute('alt')).toEqual('bar');
    expect(img.getAttribute('src')).toEqual('/bar.svg');
  });

  it('hides the logo when isMobile true', () => {
    render(
      <CmsLogo
        {...{
          isMobile: true,
          logoPosition: 'left',
          logos: [
            {
              logoAltText: 'foo',
              logoUrl: '/foo.svg',
            },
          ],
        }}
      />
    );

    const img = screen.queryByRole('img');
    expect(img).not.toBeInTheDocument();
  });
});
