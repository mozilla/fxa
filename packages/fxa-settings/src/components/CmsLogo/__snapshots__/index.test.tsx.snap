// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CmsLogo renders correctly with center positioning 1`] = `
<div>
  <img
    alt="foo"
    class="justify-center mb-4 max-h-[160px]"
    src="/foo.svg"
  />
</div>
`;

exports[`CmsLogo renders correctly with default left positioning 1`] = `
<div>
  <img
    alt="foo"
    class="justify-left mb-4 max-h-[40px]"
    src="/foo.svg"
  />
</div>
`;
